backend/tournesol/views/criteria_correlations.py

```diff
@@ -0,0 +1,83 @@
+
+import numpy as np
+from django.http import JsonResponse
+from drf_spectacular.utils import OpenApiResponse, OpenApiTypes, extend_schema
+from rest_framework import serializers
+from rest_framework.generics import GenericAPIView
+from scipy.stats import linregress
+
+from tournesol.models.poll import Poll
+from tournesol.models.ratings import ContributorRating
+
+
+def compute_correlation(scores_1: dict, scores_2: dict) -> int:
+    """
+    inputs are dictionaries with entities as keys and scores as values
+    for a given user and a given rating criteria
+    """
+
+    # At least two datapoints are necessary to compute correlations
+    if sum(entity in scores_1 for entity in scores_2) > 1:
+        h = np.corrcoef(
+            [scores_1[e] for e in scores_1 if e in scores_2],
+            [scores_2[e] for e in scores_1 if e in scores_2],
+        )
+        return None if np.isnan(h[0][1]) else h[0][1] 
+    else:
+        return None
+
+
+def compute_slope(scores_1: dict, scores_2: dict) -> int:
+    """ 
+    inputs are dictionaries with entities as keys and scores as values
+    for a given user and a given rating criteria
+    """
+
+    # At least two datapoints are necessary to compute slopes
+    if sum(entity in scores_1 for entity in scores_2) > 1:
+        return linregress(
+            [scores_1[e] for e in scores_1 if e in scores_2],
+            [scores_2[e] for e in scores_1 if e in scores_2],
+        ).slope
+    else:
+        return None
+
+
+class ContributorCriteriaCorrelationsSerializer(serializers.Serializer):
+    criterias = serializers.ListField(child=serializers.CharField(), default=list)
+    correlations = serializers.ListField(child=serializers.ListField(child=serializers.FloatField(), default=list), default=list)
+    slopes = serializers.ListField(child=serializers.ListField(child=serializers.FloatField(), default=list), default=list)
+    
+
+class ContributorCriteriaCorrelationsView(GenericAPIView):
```

Comment:
**(1)**

We can add a docstring here to display the description of the endpoint in the generated documentation.

For instance:

```python
"""Retrieve the correlation between... of the logged user, for the given poll."""
```

**(2)**

The endpoint is currently public and accessible as an anonymous user. Should it be private?

Reply:
Done 1) in fb09eb7d

I don't see the problem with 2) ... Non-authenticated seem to receive back a 401 response, isn't it? Base on your other comment, I added the test that makes sure of it.